### ISSUE TITLE
CRITICAL | App is not pulling the most recent records

### ISSUE THREAD

**Issue by apppbuilder** [FlutterFlow/flutterflow-issues#2727]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

After an update in userDocs, it doesn't show properly in the screen of the user.

It appears that the app doesn't get the new data to show it on the screen.

For example, if I change an int value from 0 to 2 in Firebase, it changes in the firebase, but keeps showing 0 in the screen, even with "rebuild all pages" being called inside an App State Update.

### Expected Behavior

As expected, if data changes in firebase, it should display correctly inside the app.

### Steps to Reproduce

1. Build a simple page that needs data from firebase.
2. Update firebase data (can be through API Call, directly from FF actions, whatever...)
3. Rebuild All Pages with Update App State.
4. Nothing changes in the screen.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4CjPLlvJFIocte7q6JcMdVoCMvNlw2RYIw1+xEVg8cDZPMC+wIZ/f7XFRuTde0dFZlfU6kgmEy0/PemvDLUe0CFxOcfLZEwcxudBfxQTmWVYiDOrqefW5/DcpTG3LE1p7QnxFAWdZtSC0Y2k+lf9y+aHbvQpTrOVcFRsiKKZCK2SrDX1iXc2URm05KZDPz

### Context

I can't do anything. The app is live, but how can the progress keep going if it's not updating the screen? Users can't do anything!

### Visual documentation

Link to the video:  [https://youtu.be/JsYqFsA-iSs](https://youtu.be/JsYqFsA-iSs)

Rules inside firebase:

![123456879](https://github.com/FlutterFlow/flutterflow-issues/assets/160440373/fa422414-cbf0-4a93-a0a0-3f7c997c7a43)

### Environment

```markdown
- FlutterFlow version: 
- Platform:
- Browser name and version:
- Operating system and version affected:
```

**Comment by rzambroni** [comment]
Hey @appbuilder, i attempted to reproduce this on a blank project but seems to be working as expected on my end. 

Please remember that making an action to update the app state won't do anything if you didn't re-make the query to firebase.

Can you reproduce this in a blank project? If so, i would greatly appreciate if you can share that code so we can take a look. I was looking at your project but there is a lot going on there and it would be better to isolate this issue so we discard any potential project specific problems.

Thanks!

**Comment by apppbuilder** [comment]
@rzambroni Hello! I'm debugging everything here. The issue is still alive. I've already reverted to some snapshots, looked at the gitHub deploys that I've made, but until now, nothing was found.

Somehow, the app can't read and is not writing properly in the user document inside users collection. Everything started overnight. I'm checking logs and everything. Soon I'll be back with something.

**Comment by apppbuilder** [comment]
@rzambroni I found the error. It's a bug, confirmed. It's not in the receiving data from Firebase, instead on the sent data to Firebase. The data is not being sent correctly, which breaks the document of the user and nothing works anymore.

I'll debrief it soon.

**Comment by apppbuilder** [comment]
@rzambroni 

TL;DR: You can close the topic. I found the bug, but managed to make it work again.

Whole explanation:

1. At the moment, the way to send JSON List to API that I use is as I explained in this post: [https://community.flutterflow.io/ask-the-community/post/is-it-only-me-or-everyone-is-having-trouble-with-google-cloud-functions-9RiymvOO9WeBK9S](https://community.flutterflow.io/ask-the-community/post/is-it-only-me-or-everyone-is-having-trouble-with-google-cloud-functions-9RiymvOO9WeBK9S)

2. Overnight, the way data was being sent to the Cloud Function changed completely.

**Correct and old way:**
![correctAndOldWay](https://github.com/FlutterFlow/flutterflow-issues/assets/160440373/81dc9b04-8570-4d93-960f-27a501ee587f)

**What was going on:**
![ThisWayIsntCorrectWay](https://github.com/FlutterFlow/flutterflow-issues/assets/160440373/4a5b6b17-43ff-4e78-9ebe-23e2de3c6aa3)

Somehow it was sending the JSON already decoded. So I tried lots of ways to send data again:

a) As it should but not working yet ([https://github.com/FlutterFlow/flutterflow-issues/issues/2252](https://github.com/FlutterFlow/flutterflow-issues/issues/2252)). **Still not working.**

b) Sending as a JSON instead of encoding into JSON. **Didn't worked.**

c) But then, coming back to string encoded and redeploying everything (without changes) worked again as shown below.

![correctWay](https://github.com/FlutterFlow/flutterflow-issues/assets/160440373/de16de99-afd4-47bb-bff6-9e42ef7be141)



`In the end.

Problem 1: The progress of the user was not being showed in the Firebase because it was sending it incorrectly.

Problem 2: I found out there was a thing in the code that wans't converting a data from the DataType/JSON correctly inside the Cloud Function and this was brokening the userDoc and causing not updating in the user screen, but only managed to see this after Problem 1 was solved.`

Thank you for your attention!

**Comment by rzambroni** [comment]
Thanks for letting us now and glad you got that solved. Have a good weekend.
